### PR TITLE
Improve performance of Related Opinions feature

### PR DESCRIPTION
**Pull request by malteos**
Added "Maximum Document Frequency" as MLT setting, and adjusted default MLT settings for performance.

### CODE REVIEW

**Comment by malteos** [comment]
Not sure why the tests are failing but I don't see how my changes should cause this:

```
 INFO:cl.lib.command_utils:Successfully added doc 81: Richard Johnson v. State of Indiana
INFO:cl.lib.command_utils:Adding new document found at: test/search/opinion_text.txt
ERROR:celery.app.trace:Task cl.scrapers.tasks.extract_doc_content[41bd461e-3980-4e31-b056-b060c86c7bcf] raised unexpected: AttributeError("'module' object has no attribute 'Site'",)
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/celery/app/trace.py", line 385, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/opt/courtlistener/cl/scrapers/tasks.py", line 311, in extract_doc_content
    update_document_from_text(opinion)
  File "/opt/courtlistener/cl/scrapers/tasks.py", line 243, in update_document_from_text
    site = get_scraper_object_by_name(court)
  File "/opt/courtlistener/cl/lib/juriscraper_utils.py", line 22, in get_scraper_object_by_name
    return importlib.import_module(full_module_path).Site()
AttributeError: 'module' object has no attribute 'Site'
```

EDIT: I see... the error above did cause the build the fail.

File: cl/lib/search_utils.py [hunk 1]
```diff
@@ -989,16 +989,21 @@ def get_related_clusters_with_cache(cluster, request):
         for item in sub_opinion_queries:
             sub_opinion_query |= item
 
+        # Set MoreLikeThis parameters
+        # (see https://lucene.apache.org/solr/guide/6_6/other-parsers.html#OtherParsers-MoreLikeThisQueryParser)
+        mlt_params = dict(
```
Comment (mlissner) [inline cl/lib/search_utils.py:994]: Why use dict here? Generally it's better just to use `{}`.

File: cl/search/tests.py [hunk 2]
```diff
@@ -618,7 +616,12 @@ def test_issue_727_doc_att_numbers(self):
         self.assertEqual(r.status_code, HTTP_200_OK)
 
 
-@override_settings(RELATED_USE_CACHE=False,)
+@override_settings(
```
Comment (mlissner) [inline cl/search/tests.py:619]: Can you put a comment here as to why it's important to override these for the test? Looks like you're...disabling them, maybe?

File: cl/lib/search_utils.py (hunk 1 above)
Comment (malteos) [reply cl/lib/search_utils.py]: I was just too lazy to write the `"` ;-)

File: cl/search/tests.py (hunk 2 above)
Comment (malteos) [reply cl/search/tests.py:619]: I added a comment (default settings are optimized for run-time performance and limit the search space)

**Comment by malteos** [comment]
Thanks! I updated the PR according to your feedback.

**Pull request merged**